src/Forms/FieldList.php

```diff
@@ -77,18 +120,67 @@ public function dataFields()
      */
     public function saveableFields()
     {
-        if (!$this->sequentialSaveableSet) {
-            $this->collateDataFields($this->sequentialSaveableSet, true);
+        if (empty($this->sequentialSaveableSet)) {
+            $fields = [];
+            $this->recursiveWalk(function (FormField $field) use (&$fields) {
+                if (!$field->canSubmitValue()) {
+                    return;
+                }
+                $name = $field->getName();
+                if (isset($fields[$name])) {
+                    $this->fieldNameError($field, __FUNCTION__);
+                }
+                $fields[$name] = $field;
+            });
+            $this->sequentialSaveableSet = $fields;
         }
         return $this->sequentialSaveableSet;
     }
 
+    /**
+     * Return array of all field names
```

Comment:
Actually, only data field names. Should this method be `dataFieldNames()` ? Or should $dataFieldsOnly be a flag maybe? Up to you. :)

Reply:
tbh, the only reason I've added this is because it was requested here: https://github.com/silverstripe/silverstripe-framework/pull/6472#issuecomment-272305378

I'm not sure it adds a huge amount of value - do you think it's worth adding?